internal/controller/accountiam_controller.go

```diff
@@ -287,26 +334,67 @@ func (r *AccountIAMReconciler) reconcileOperandResources(ctx context.Context, in
 		}
 	}
 
+	// Temporary update issuer in platform-auth-idp configmap
+	klog.Infof("Updating platform-auth-idp configmap")
+	idpconfig := &corev1.ConfigMap{}
+	if err := r.Get(ctx, client.ObjectKey{Name: "platform-auth-idp", Namespace: instance.Namespace}, idpconfig); err != nil {
+		klog.Errorf("Failed to get configmap platform-auth-idp in namespace %s", instance.Namespace)
+		return err
+	}
+	idpconfig.Data["OIDC_ISSUER_URL"] = decodedData.DefaultIDPValue
+	if err := r.Update(ctx, idpconfig); err != nil {
+		klog.Errorf("Failed to update ConfigMap platform-auth-idp in namespace %s: %v", instance.Namespace, err)
+		return err
+	}
+
+	// Delete the platform-auth-service and platform-identity-provider pod to restart it
+	if err := r.restartAndCheckPod(ctx, instance.Namespace, "platform-auth-service"); err != nil {
+		return err
+	}
+	klog.Infof(" platform-auth-service pod is ready")
+
+	if err := r.restartAndCheckPod(ctx, instance.Namespace, "platform-identity-provider"); err != nil {
+		return err
+	}
+	klog.Infof(" platform-identity-provider pod is ready")
+
+	klog.Infof("MCSP operand resources created successfully")
```

Comment:
Could we check the if the ConfigMap has been updated with expected `OIDC_ISSUER_URL`?

If the ConfigMap has been updated before, we could skip restarting the IM pod.